### ISSUE TITLE
Not working on Firefox

### ISSUE THREAD

**Issue by williamclavier** [arc298/tiktok-captcha-solver#1]
I am not able to get captchas to pop up on chrome as chrome is detected as being controlled by software so the site as a whole doesn't work. That is why I attempted to use firefox where the site worked with no issues. The only issue was that the captcha solver doesn't work for some reason and I was wondering if you could help me or look into it yourself.
Thanks

**Comment by arc298** [comment]
I'm observing that the captcha is solved correctly on Firefox when you are redirected to the standalone, verify page. I'm not able to reproduce the captcha appearing while on a normal TikTok page on Firefox. I would suggest turning on `headless: false` when creating your browser instance and observing what happens.

**Comment by tstrg** [comment]
I'm facing the same issue with firefox. I can see the _watchForCaptchaAdd throws following exception

page.waitForSelector: Timeout 5000ms exceeded.
=========================== logs ===========================
waiting for selector "[id$="verify-ele"]" to be visible

**Comment by duke1921** [comment]
> I'm observing that the captcha is solved correctly on Firefox when you are redirected to the standalone, verify page. I'm not able to reproduce the captcha appearing while on a normal TikTok page on Firefox. I would suggest turning on `headless: false` when creating your browser instance and observing what happens.

Hey,  I've been testing it using puppeteer. And I think the OP wasn't able to explain the issue properly. Let me rephrase it, as I'm also getting that same, on chrome as well. See, it works fine everywhere on the "standalone verify page". But the problem comes when the captcha comes when performing some actions in later steps without the "page load/reload". (For example like, logging in after form fill or sign up after form fill, etc. In those cases, captcha comes as soon as we press the submit button, even before loading anything for the next page.) If you can solve that asap, then man put up your "Buy Me a Coffee" Link, I'll surely buy you some, as I need this for a small project that I'm working on right now And time is of the essence for me currently. And thanks a lot!

**Comment by tstrg** [comment]
I use captcha solver with Firefox on the login page:

`https://www.tiktok.com/login/?lang=en&enter_method=click_top_bar&redirect_url=https%253A%252F%252Fwww.tiktok.com%252F&enter_from=homepage_hot&is_modal=1&hide_close_btn=1`

When I modify CaptchaSolver.js and changed

> erifyElement: '[id$="verify-ele"]', 

to

> verifyElement: '[id$="login_slide"]',

Everything works as expected.

**Comment by arc298** [comment]
@tstrg I'm having trouble reproducing it where it an element with id login_slide exists in Firefox. Potentially there could be other variations of the slider markup that I haven't seen. Could you take a screenshot of what it looks like?

**Comment by tstrg** [comment]
@arc298 Steps to reproduce:

1. Visit ["login url"](https://www.tiktok.com/login/?lang=en&enter_method=click_top_bar&redirect_url=https%253A%252F%252Fwww.tiktok.com%252F&enter_from=homepage_hot&is_modal=1&hide_close_btn=1)
2. Click on "Use phone / email / username"
3. Click on "Log in with email or username"
4. Enter email and password
5. Click on "Log in"
6. Captcha appears here

Screenshot and code example attached below.

![image](https://user-images.githubusercontent.com/22916452/123814342-e970c100-d8f5-11eb-8fe9-8b60836ebe82.png)

```
 const { firefox} = require('playwright'); 
 const CaptchaSolver = require('tiktok-captcha-solver');

 (async () => {

  const browser = await firefox.launch({headless: false}); //lets watch what it's doing
  const context = await browser.newContext(); 
  const page = await context.newPage();
  // the captcha solver should be initialized before visiting the page
  const captchaSolver = new CaptchaSolver(page);   

  try{
   await page.goto('https://www.tiktok.com/login/?lang=en&enter_method=click_top_bar&redirect_url=https%253A%252F%252Fwww.tiktok.com%252F&enter_from=homepage_hot&is_modal=1&hide_close_btn=1');
   
   const userEmail = "your_email"; // use your email
   const userPassword = `your_password`; // user your password

      await page.click('//div[contains(text(),"email")]',{delay: 2000,timeout: 5000}); // Select email method      
      await page.hover('//a[contains(text(),"email")]',{delay: 2000,timeout: 5000});
      await page.click('//a[contains(text(),"email")]',{delay: 1000,timeout: 5000}); // Switch login form to email options
      await page.type('input[name=email]',userEmail,{ delay: 200,timeout: 5000});
      await page.type('input[name=password]',userPassword,{ delay: 200, timeout: 5000 });
      await page.hover('button[type=submit]',{timeout:300});
      await page.click('button[type=submit]');
      await page.waitForSelector("text=Verification:",{ timeout: 5000 });
      // solve the captcha
      await captchaSolver.solve();
      await page.waitForSelector('text=Login success', {state: "attached",timeout:5000}); 
      await browser.close();
  } catch (e) {
    console.error(e);
    await browser.close();
  }
 })();
```

**Comment by arc298** [comment]
@tstrg Thanks for the steps to reproduce and code sample. This is fixed in v0.0.4.
